src/script/components/Modals/DetailViewModal/index.tsx

```diff
@@ -234,16 +246,91 @@ const DetailViewModal: FC<DetailViewModalProps> = ({
   }, []);
 
   const modalId = 'detail-view';
+
   useEffect(() => {
     const onKeyDown = (event: KeyboardEvent): void => {
       preventFocusOutside(event, modalId);
     };
+
     document.addEventListener('keydown', onKeyDown);
+
     return () => {
       document.removeEventListener('keydown', onKeyDown);
     };
   }, []);
 
+  const onButtonClick = (event: React.MouseEvent<HTMLElement>) => {
```

Comment:
Let's try to extract that entire functionality into its own component that will wrap the `<img>` tag, WDYT? 
This will leave the `DetailViewModal` component rather small and readable